### ISSUE TITLE
Additional OpenAPI field properties: title and readOnly

### ISSUE THREAD

**Issue by jeff00seattle** [stripe/openapi#142]
Request adding OpenAPI field properties:

- **title** -- shorter wording over existing field property **description**
- **readOnly** -- to determine if a Stripe API client can update the field.

**Comment by pakrym-stripe** [comment]
Hi,

> readOnly

There are separate schemas for resources and update parameters, you can roughly deduce what resource properties are writeable to examining an update method body.

> title

Unfortunately, I do not think this is going to happen anytime soon. The API surface is very large and manually adding titles to everything is quite a bit of work. 

Can you share your use case for these two sets of values?

**Comment by jeff00seattle** [comment]
@pakrym-stripe 

> There are separate schemas for resources and update parameters, you can roughly deduce what resource properties are writeable to examining an update method body.

Please, I would be highly interested, to clarify where are these **separate schemas for resources and update parameters**.

Much appreciate your response.

**Comment by jeff00seattle** [comment]
@pakrym-stripe 

> Can you share your use case for these two sets of values?

Working on an integration service that requires providing fields per object model (schema). The minimal field properties requirement is to provide usage expectations:
- `type`
- `title` -- The available field property `description` is too verbose, whereas `title` would be short and to the point.
- `readOnly`

Example a field `facebookfans` minimally using these three field properties
```
      "facebookfans": {
        "title": "Facebook Fans",
        "readOnly": false,
        "type": "number"
      },
```

Salesforce API schema provides:
```
{ ...subSchema, title: field.label, readOnly: !field.updateable }
```

HubSpot API schema provides:
```
{ ...subSchema, title: field.label, readOnly: field.modificationMetadata.readOnlyValue }
```


Both Salesforce


        ExpectedRevenue: {
          type: 'number',
          minimum: 0,
          maximum: 10000000000000000,
          exclusiveMaximum: true,
          multipleOf: 0.01,
          title: 'Expected Amount',
          readOnly: true,
        },

**Comment by pakrym-stripe** [comment]
>Please, I would be highly interested, to clarify where are these separate schemas for resources and update parameters.

Let's the the account resource. The schema for the resource itself is:
```yaml
components:
  schemas:
    account:
      description: >-
        This is an object representing a Stripe account. You can retrieve it to ...
      properties:
        business_profile:
          anyOf:
            - $ref: '#/components/schemas/account_business_profile'
          description: Business information about the account.
          nullable: true
        business_type:
          description: The business type.
          enum:
            - company
            - government_entity
            - individual
            - non_profit
          nullable: true
          type: string
          x-stripeBypassValidation: true
        capabilities:
          $ref: '#/components/schemas/account_capabilities'
```

The schema for the account update parameters:

```
'/v1/accounts/{account}':
    post:
      description: >-
        <p>Updates a <a href="/docs/connect/accounts">connected account</a> by
        setting the values of the parameters passed. Any parameters not provided
        are

        left unchanged.</p>


        <p>For Custom accounts, you can update any information on the account.
        For other accounts, you can update all information until that

        account has started to go through Connect Onboarding. Once you create an
        <a href="/docs/api/account_links">Account Link</a>

        for a Standard or Express account, some parameters can no longer be
        changed. These are marked as <strong>Custom Only</strong> or
        <strong>Custom and Express</strong>

        below.</p>


        <p>To update your own account, use the <a
        href="https://dashboard.stripe.com/settings/account">Dashboard</a>.
        Refer to our

        <a href="/docs/connect/updating-accounts">Connect</a> documentation to
        learn more about updating accounts.</p>
      operationId: PostAccountsAccount
      parameters:
        - in: path
          name: account
          required: true
          schema:
            maxLength: 5000
            type: string
          style: simple
      requestBody:
        content:
          application/x-www-form-urlencoded:
            encoding:
              business_profile:
                explode: true
                style: deepObject
              capabilities:
                explode: true
                style: deepObject
              company:
                explode: true
                style: deepObject
              documents:
                explode: true
                style: deepObject
              expand:
                explode: true
                style: deepObject
              individual:
                explode: true
                style: deepObject
              metadata:
                explode: true
                style: deepObject
              settings:
                explode: true
                style: deepObject
              tos_acceptance:
                explode: true
                style: deepObject
            schema:
              additionalProperties: false
              properties:
                account_token:
                  description: >-
                    An [account
                    token](https://stripe.com/docs/api#create_account_token),
                    used to securely provide details to the account.
                  maxLength: 5000
                  type: string
                business_profile:
                  description: Business information about the account.
                  properties:
                    mcc:
                      maxLength: 4
                      type: string
                    monthly_estimated_revenue:
                      properties:
                        amount:
                          type: integer
                        currency:
                          type: string
                      required:
                        - amount
                        - currency
                      title: monthly_estimated_revenue_specs
                      type: object
                    name:
                      maxLength: 5000
                      type: string
                    product_description:
                      maxLength: 40000
                      type: string
```

**Comment by jeff00seattle** [comment]
@pakrym-stripe 
Merry Christmas

Appreciate your response
> The schema for the resource itself...
Is there a way to determine which parameters are **readOnly**?

> The schema for the account update parameters:
There appears to be a web scraping of the HubSpot API documentation.
Is there another way to derive update parameters via another JSON file similar to **spec3.json**?

**Comment by pakrym-stripe** [comment]
>Is there a way to determine which parameters are readOnly?

If the parameter is in the request schema then it's writable.

**Comment by richardm-stripe** [comment]
I'm closing this ticket as "wontfix", as `readOnly` on a resource schema doesn't really apply to our API (as @pakrym-stripe has described) and it's unlikely that we provide "title" without more rationale/demand. We are subscribed to the issue so feel free to continue asking question -- or if anybody else reading this has a use case for these, please feel free to comment and describe.
